### PR TITLE
libvirt: debug log image cache download progress

### PR DESCRIPTION
**Pull request by dcbw**
@abhinavdahiya

### CODE REVIEW

File: pkg/tfvars/libvirt/cache.go [hunk 1]
```diff
@@ -102,6 +102,26 @@ func cacheKey(etag string) (key string, err error) {
...
 
+type progressWatcher struct {
+	total    uint64
+	oldTotal uint64
+}
+
+func (w *progressWatcher) Write(p []byte) (int, error) {
+	n := len(p)
+	w.total += uint64(n)
+	// Log progress every 3 MiB
```
Comment (abhinavdahiya) [inline pkg/tfvars/libvirt/cache.go]: the image size as of today is around 700 MB, for each 3MB there's going to be 233 debug lines.... that's not acceptable.

**Comment by abhinavdahiya** [comment]
Installer downloads the libvirt image very rarely, as the boot-images don't matter very much, So I don't think the progress bread crumbs provide a lot of value.

**Comment by dcbw** [comment]
> Installer downloads the libvirt image very rarely, as the boot-images don't matter very much, So I don't think the progress bread crumbs provide a lot of value.

@abhinavdahiya I added this because of the download timeout issues I encountered; I found it useful to know much earlier than the TCP timeout when the image download had stalled while tailing the install log to track progress.

File: pkg/tfvars/libvirt/cache.go (hunk 1 above)
Comment (dcbw) [reply pkg/tfvars/libvirt/cache.go]: @abhinavdahiya How about a time-based approach, where log messages are printed every 5 seconds IFF > 3MiB has been read?

Comment (abhinavdahiya) [reply pkg/tfvars/libvirt/cache.go]: Anything more than an couple of lines for download progress is _imo_ more than what's acceptable.

**Comment by abhinavdahiya** [comment]
> I found it useful to know much earlier than the TCP timeout when the image download had stalled while tailing the install log to track progress.

hmm, not sure how it was helpful for you to know at what % timeout happened for you.? the installer doesn't resume the download...

**Pull request closed without merging**